api/src/main/resources/workbench.yaml

```diff
@@ -972,6 +1079,21 @@ definitions:
         REVIEW_ID_VERIFICATION
     ]
 
+  ConceptSetDomain:
```

Comment:
How does this interact with the dynamic domain CSV file which feeds into the CDR indexing pipeline? The list here doesn't match those.

https://github.com/all-of-us/workbench/blob/master/api/db-cdr/csv/domain.csv

Reply:
This table matches the cohort tables in our CDM schema that have standard / source concepts:

https://github.com/all-of-us/workbench/blob/master/api/config/cdm/cdm_5_2.json

Basically we're giving you all the tables you can pull stuff from when using a concept set.

The domains in the CSV file / OMOP data map to this, many-to-many... currently represented in this mapping:

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/api/ConceptsController.java#L25

and when searching for concepts we will need to allow for multiple "domain" values in the OMOP data for a single "domain" indicated here.

We could pick a different term instead of "domain" for this, but I'm not thinking of a good one. 

Basically OMOP has concept -> domain -> one or more OMOP tables; they could have decided to instead do a many-to-many from concepts to OMOP tables, which is kind of what we're doing.

Reply:
I'm not entirely following - is this at all related to this field here? https://github.com/all-of-us/workbench/blob/f8eb2419cb5d3a97c6d9db630e56775689dfaee3/api/src/main/resources/client_api.yaml#L375

Are tables containing person_ids the only tables relevant to the cohort builder?

Reply:
Thanks for pointing that enum out! It is used in concept search -- when a user asks for say "OBSERVATION" we look up in ConceptsController.DOMAIN_MAP the set of OMOP domain IDs that it maps to, and query for just those concepts.

I switched ConceptSet to use it... with logic enforcing that certain values (GENDER, RACE, and ETHNICITY) aren't valid. (These aren't like the domains that correspond to separate OMOP tables, and we don't think we need to support concept sets for them.) I like reusing the enum better than maintaining two enums I think.

I also added validation logic to ensure that concept IDs being added must have an OMOP domain ID that is mapped to by the domain on the concept set.